musician/tests/test_musician_api.py

```diff
@@ -0,0 +1,129 @@
+import datetime
+
+from django.test import TestCase
+from django.urls import reverse
+from rest_framework import status
+
+from rest_framework.test import APIClient
+
+from musician.models import Musician
+from musician.serializers import MusicianSerializer
+
+MUSICIAN_URL = reverse("musician:manage-list")
+
+
+class MusicianApiTests(TestCase):
+    def setUp(self) -> None:
+        self.client = APIClient()
+        Musician.objects.create(
+            first_name="Joseph",
+            last_name="Green",
+            instrument="the guitar",
+            age=17,
+        )
+        Musician.objects.create(
+            first_name="Robert",
+            last_name="Brown",
+            instrument="the guitar",
+            age=28,
+        )
+
+    def test_age_has_min_value(self):
+        payload = {
+            "first_name": "Jim",
+            "last_name": "Greg",
+            "instrument": "the violin",
+            "age": 10
+        }
+
+        response = self.client.post(MUSICIAN_URL, payload)
+        musicians = Musician.objects.all()
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(musicians.count(), 2)
+
+    def test_get_musicians(self):
+        musicians = self.client.get(MUSICIAN_URL)
+        serializer = MusicianSerializer(Musician.objects.all(), many=True)
+        self.assertEqual(musicians.status_code, status.HTTP_200_OK)
+        self.assertEqual(musicians.data, serializer.data)
+
+    def test_post_musicians(self):
+        musicians = self.client.post(
+            MUSICIAN_URL,
+            {
+                "first_name": "Bob",
+                "last_name": "Yellow",
+                "instrument": "the guitar",
+                "age": 47
+            },
+        )
+        db_musicians = Musician.objects.all()
+        self.assertEqual(musicians.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(db_musicians.count(), 3)
+        self.assertEqual(db_musicians.filter(first_name="Bob").count(), 1)
+
+    def test_get_musician(self):
+        response = self.client.get(f"{MUSICIAN_URL}1/")
+        serializer = MusicianSerializer(
+            Musician(
+                id=1,
+                first_name="Joseph",
+                last_name="Green",
+                instrument="the guitar",
+                age=17,
+                date_of_applying=datetime.date.today(),
+            )
```

Comment:
also can send here Musician.objects.first()

Reply:
Also here use that user from setUp